### PR TITLE
Wrap up of the last best setting we had on aug 10

### PR DESCRIPTION
**Pull request by MMDPROJECT**
Changes in this wrapping includes:
1. Directory Creation of the Project:
1.1. Put `create_proj_dirs` in a separate .py for better modulation.
1.2. Separated checkpoints that correspond to best evaluation scores we get in a separate folder `best_checkpoints_dirname` from ordinary checkpoints. 
1.3. Better naming of the saved files in some cases; e.g. added mean reward to the best saved models .

2. Completion of `requirements.txt` + Removing unnecessary imports in the files.

3. Improve logging:
3.1. Removed `scaling factor` in `train.py`.
3.2. Replaced _Frame_ by _Iteration_ in the logs for consistent naming.

4. Testing: Fix default wrappers and settings used for initializtion of test.

**NOTE:**
All of the changes are applied on top of the settings we had by the end of _Aug 10_. More specifically the didn't have frame skipping fix, Change of hyperparameters of the optimizer to something other than what's provided in the paper, or applying gradient clipping; those are applied on the test branches

### CODE REVIEW

**Review by Retriever5500 — APPROVED** [review]

**Comment by Retriever5500** [comment]
LGTM
